### ISSUE TITLE
Bot responds to messages clearly not meant for it

### ISSUE THREAD

**Issue by abyss** [jordanclarkedev/bitdicebot#5]
The bot responds to messages that match it's requirements (leading ! and some numbers, I think?) which is resulting in some funny problems (and may be related to #4 's crashing).

You likely need to process the message further to make sure it matches the exact intents of the bot.

Here's some screenshots demonstrating, from my server. Both of these messages were intended for Tatsumaki bot, I believe. The first also demonstrates a really funny situation - I believe the bot is seeing a user tag, which contains numbers, and processing out the number in it. 

![image](https://user-images.githubusercontent.com/16959744/80397870-d216ed80-8884-11ea-8948-9806201e4600.png)

![image](https://user-images.githubusercontent.com/16959744/80397919-e3f89080-8884-11ea-9abc-cdc84654471b.png)

**Comment by abyss** [comment]
Spit balling a bit -

Supporting the mixed syntax you allow for rolls (`! 2` and `! 2 r` for example) makes solving this complicated. 

In effect, you're treating the bot's prefix (`!`) as the command, and passing the rest to the command. Typically, there would be a command following the prefix that allows the bot to verify the message is intended for them (prefix+command matching, ignore if not). 

To change your bot to have a command as well (for example, `!roll`) would result in changing the way your bot does the primary thing your bot does (that is, rolling dice!). Is that something you're willing to do?

If so, the fix seems more simple, but users will need to be notified (and hopefully figure it out). 

If not, the fix is more complicated (process the entire message to determine if it's a valid message with all possible combinations?), but users can expect the exact same behavior as before, except for the unintended behavior.

**Comment by jordanclarkedev** [comment]
Yeah, I wasn't sure what other bots this would/could conflict with.

Apologies for not noticing this issue posted earlier.

I'll temporarily change it such that the integer has to immediately follow the !, and that the resist flag has to be after that. 
That should fix the weird glitch with user IDs and avoid changing the syntax _too_ much.

I'll leave this open for now, but the wide majority of bot conflicts should be cleared up as of 1.1.4

**Comment by bendoerr** [comment]
I really think that any bot that does not support custom prefixes should attempt to default to a prefix that would likely be unused by other bots. For example this bot could default to the prefix `!bitd` and be used for example `!bitd 2` or `!bitd 2r`.

**Comment by abyss** [comment]
I agree - but I personally prefer making all bots able to have changed prefixes. That's a longer term goal though.

Given this bots unique-ish syntax, I think a good workaround for now is making it stricter on requiring a sensible message. Collisions past that point are much more easily mitigated on the permissions side.
